### PR TITLE
[windows] Add Windows AppLocker Data Stream (Packaged app-Deployment)

### PR DESCRIPTION
**Pull request by nicpenning**
- Enhancement

## What does this PR do?

Adding the Microsoft Windows AppLocker Packaged app-Deployment channel to the Windows Integration

Resolves Part of - https://github.com/elastic/integrations/issues/6979

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Comment by nicpenning** [comment]
This should be ready for testing. Please advise!

File: packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      if: ctx.winlog?.time_created != null
+      on_failure:
+        - remove:
+            field: winlog.time_created
+            ignore_failure: true
+        - append:
+            field: error.message
+            value: "fail-{{{ _ingest.on_failure_processor_tag }}}"
+        - fail:
+            message: "Processor {{ _ingest.on_failure_processor_type }} with tag {{ _ingest.on_failure_processor_tag }} in pipeline {{ _ingest.on_failure_pipeline }} failed with message: {{ _ingest.on_failure_message }}"
```
Comment (efd6) [inline packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml:29]:
```suggestion
            message: "Processor {{{ _ingest.on_failure_processor_type }}} with tag {{{ _ingest.on_failure_processor_tag }}} in pipeline {{{ _ingest.on_failure_pipeline }}} failed with message: {{{ _ingest.on_failure_message }}}"
```
(similar throughout)

File: packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      field: event.type
+      value: start
+  - convert:
+      field: winlog.record_id
+      type: string
+      ignore_failure: true
+      ignore_missing: true
+
+  ## User fields.
+
```
Comment (efd6) [inline packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml:50]:
```suggestion
```

File: packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+      ignore_empty_value: true
+      if: ctx._temp?.user_parts != null && ctx._temp.user_parts.size() == 2
+  - rename:
+      field: winlog.user.name
+      target_field: user.name
+      ignore_failure: true
+      ignore_missing: true
+      if: ctx.user?.name == null
+
+    ## User data fields.
```
Comment (efd6) [inline packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml:80]:
```suggestion
  ## User data fields.
```

File: packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            ignore_failure: true
+
+    ## Fully Qualified Binary Name (FQBN) / File Fields
+  - grok:
+      field: winlog.user_data.Fqbn
+      ignore_missing: true
+      patterns:
+      - "^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}L=(?<file.x509.subject.locality>.*),%{SPACE}S=(?<file.x509.subject.state_or_province>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\\\]*)\\\\(?<file.pe.product>[^\\\\]*)\\\\(?<file.pe.original_file_name>[^\\\\]*)\\\\(?<file.pe.file_version>.*)$"
+      - "^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}L=(?<file.x509.subject.locality>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\\\]*)\\\\(?<file.pe.product>[^\\\\]*)\\\\(?<file.pe.original_file_name>[^\\\\]*)\\\\(?<file.pe.file_version>.*)$"
+      - "^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}S=(?<file.x509.subject.state_or_province>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\\\]*)\\\\(?<file.pe.product>[^\\\\]*)\\\\(?<file.pe.original_file_name>[^\\\\]*)\\\\(?<file.pe.file_version>.*)$"
```
Comment (efd6) [inline packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml:153]:
```suggestion
      - '^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}L=(?<file.x509.subject.locality>.*),%{SPACE}S=(?<file.x509.subject.state_or_province>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\]*)\\(?<file.pe.product>[^\\]*)\\(?<file.pe.original_file_name>[^\\]*)\\(?<file.pe.file_version>.*)$'
      - '^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}L=(?<file.x509.subject.locality>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\]*)\\(?<file.pe.product>[^\\]*)\\(?<file.pe.original_file_name>[^\\]*)\\(?<file.pe.file_version>.*)$'
      - '^CN=(?<file.x509.subject.common_name>.*),%{SPACE}O=(?<file.x509.subject.organization>.*),%{SPACE}S=(?<file.x509.subject.state_or_province>.*),%{SPACE}C=(?<file.x509.subject.country>[^\\]*)\\(?<file.pe.product>[^\\]*)\\(?<file.pe.original_file_name>[^\\]*)\\(?<file.pe.file_version>.*)$'
```
(let's not proliferate `\`)

File: packages/windows/data_stream/applocker_packaged_app_deployment/manifest.yml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        description: Go Duration syntax (eg. 10s)
+        show_user: true
+        required: true
+        default: 10s
+      - name: search
+        type: text
+        title: Splunk search string
+        show_user: false
+        required: true
+        default: "search sourcetype=\"XmlWinEventLog:Microsoft-Windows-AppLocker/Packaged app-Deployment\""
```
Comment (efd6) [inline packages/windows/data_stream/applocker_packaged_app_deployment/manifest.yml:75]:
```suggestion
        default: 'search sourcetype="XmlWinEventLog:Microsoft-Windows-AppLocker/Packaged app-Deployment"'
```

File: packages/windows/data_stream/applocker_packaged_app_deployment/_dev/test/pipeline/test-events-microsoft-windows-applocker-packaged-app-deployment.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                    "TargetUser": "S-1-5-21-1133191089-1850170202-1535859923-200319",
+                    "TargetProcessId": "4584",
+                    "PackageLength": "18",
+                    "Package": "Microsoft.BingNews",
+                    "FqbnLength": "118",
+                    "Fqbn": "CN=MICROSOFT CORPORATION, O=MICROSOFT CORPORATION, L=REDMOND, S=WASHINGTON, C=US\\MICROSOFT.BINGNEWS\\APPX\\4.55.62231.00"
+                },
+                "event_id": "8023",
+                "level": "Information",
+                "opcode": "Info\u0000",
```
[the remark this answers is not in the document]
Comment (nicpenning) [reply packages/windows/data_stream/applocker_packaged_app_deployment/_dev/test/pipeline/test-events-microsoft-windows-applocker-packaged-app-deployment.json:38]: In an automated way, I will tune and address accordingly. Good catch!

File: packages/windows/data_stream/applocker_packaged_app_deployment/agent/stream/httpjson.yml.hbs [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+{{#unless username}}
+{{#unless password}}
+{{#if token}}
+  - set:
+      target: header.Authorization
+      value: {{token}}
+{{/if}}
+{{/unless}}
+{{/unless}}
+response.decode_as: application/x-ndjson
```
[the remark this answers is not in the document]
Comment (nicpenning) [reply packages/windows/data_stream/applocker_packaged_app_deployment/agent/stream/httpjson.yml.hbs:49]: Strange,  I copied from the previous integration we merged, I will investigate.

File: packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml (hunk 4 above)
Comment (nicpenning) [reply packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml:152]: This makes sense. Curious to look back at the other integrations to see how we handled these.

File: packages/windows/data_stream/applocker_packaged_app_deployment/manifest.yml (hunk 5 above)
Comment (nicpenning) [reply packages/windows/data_stream/applocker_packaged_app_deployment/manifest.yml:75]: I will tune and address accordingly.

File: packages/windows/data_stream/applocker_packaged_app_deployment/agent/stream/httpjson.yml.hbs (hunk 7 above)
[the remark this answers is not in the document]
Comment (nicpenning) [reply packages/windows/data_stream/applocker_packaged_app_deployment/agent/stream/httpjson.yml.hbs:49]: Confirmed, this is in other datastreams before my time 😀

https://github.com/nicpenning/integrations/blob/db66026938161da65ba7b527d8111ede6e88ec3d/packages/windows/data_stream/sysmon_operational/agent/stream/httpjson.yml.hbs#L49

**Comment by nicpenning** [comment]
Cleaned up null terminations, please test.

File: packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+      ignore_empty_value: true
+      if: ctx.winlog?.user_data?.FileHash != "-"
+      
+on_failure:
+  - set:
+      field: event.kind
+      value: pipeline_error
+  - append:
+      field: error.message
+      value: "{{ _ingest.on_failure_message }}"
```
Comment (efd6) [inline packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml:174]: And here.

File: packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+            value: "fail-{{{ _ingest.on_failure_processor_tag }}}"
+        - fail:
+            message: "Processor {{{ _ingest.on_failure_processor_type }}} with tag {{{ _ingest.on_failure_processor_tag }}} in pipeline {{{ _ingest.on_failure_pipeline }}} failed with message: {{{ _ingest.on_failure_message }}}"
+
+  - set:
+      field: event.kind
+      value: event
+  - set:
+      field: event.code
+      value: '{{winlog.event_id}}'
```
[the remark this answers is not in the document]
Comment (nicpenning) [reply packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml:36]: What are you suggesting here and the other areas for the pipeline?

[the remark this answers is not in the document]
Comment (nicpenning) [reply packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml:36]: Nevermind, I see this is related to the additional braces as commented before. I will address.

**Comment by nicpenning** [comment]
All findings corrected. Please test!

File: packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (efd6) [reply packages/windows/data_stream/applocker_packaged_app_deployment/elasticsearch/ingest_pipeline/default.yml]: Yup

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**